doc/source/ray-air/examples/batch_forecasting.ipynb

```diff
@@ -0,0 +1,1444 @@
+    "\n",
+    "    # Load data.\n",
+    "    df_list = [read_data(f, sample_location_id) for f in s3_files]\n",
+    "    df_raw = pd.concat(df_list, ignore_index=True)\n",
+    "    # Abort Tune to avoid Tune Error if df has too few rows\n",
+    "    if df_raw.shape[0] < FORECAST_LENGTH:\n",
+    "        print_time(\n",
+    "            f\"Location {sample_location_id} has only {df_raw.shape[0]} rows\"\n",
+    "        )\n",
+    "        return None\n",
+    "\n",
+    "    # Transform data.\n",
+    "    df = transform_df(df_raw)\n",
+    "    # Abort Tune to avoid Tune Error if df has too few rows\n",
+    "    if df.shape[0] < FORECAST_LENGTH:\n",
+    "        print_time(f\"Location {sample_location_id} has only {df.shape[0]} rows\")\n",
+    "        return None\n",
+    "\n",
+    "    # Train model.\n",
+    "    if model_type == \"arima\":\n",
+    "        train_df, test_df = train_test_split(df, test_size=FORECAST_LENGTH)\n",
+    "        model = StatsForecast(df=train_df, models=model, freq=\"D\")\n",
+    "        # Inference model and evaluate error.\n",
+    "        error, future = evaluate_model_statsforecast(model, test_df)\n",
+    "    else:\n",
+    "        # model type is Prophet\n",
+    "        model = model.fit(df[[\"ds\", \"y\"]])\n",
+    "        # Inference model and evaluate error.\n",
+    "        error, future = evaluate_model_prophet(model)\n",
+    "\n",
+    "    # Define a model checkpoint using AIR API.\n",
+    "    # https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html\n",
+    "    checkpoint = ray.air.checkpoint.Checkpoint.from_dict(\n",
+    "        {\n",
+    "            \"model\": model,\n",
+    "            \"forecast_df\": future,\n",
+    "            \"location_id\": sample_location_id,\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "    # Save checkpoint and report back metrics, using ray.air.session.report()\n",
+    "    # The metrics you specify here will appear in Tune summary table.\n",
+    "    # They will also be recorded in Tune results under `metrics`.\n",
+    "    metrics = dict(error=error)\n",
+    "    session.report(metrics, checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1619c7d",
+   "metadata": {},
+   "source": [
+    "## Run batch training on Ray Tune <a class=\"anchor\" id=\"run_tune_search\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "642d21d4",
+   "metadata": {},
+   "source": [
+    "**In the cell below, we configure the resources allocated per trial.** \n",
+    "\n",
+    "Tune uses this resources allocation to control the parallelism. For example, if each trial was configured to use 4 CPUs, and the cluster had only 32 CPUs, then Tune will limit the number of concurrent trials to 8 to avoid overloading the cluster. For more information, see [A Guide To Parallelism and Resources](https://docs.ray.io/en/master/tune/tutorials/tune-resources.html#tune-parallelism)."
+   ]
```

Comment:
We can skip this section (or put it in the bottom); it's not critical path for the user to know